src/boost_histogram/_internal/hist.py

```diff
@@ -23,6 +24,7 @@
 )
 
 import numpy as np
+from pytest import approx
```

Comment:
We can't use pytest inside the library itself. There are tools for this like `np.allclose`. Also `compare` should return True/False, not errors / asserts. Let's move the changes to hist.py into a different PR, as that's not related to using `approx` in the tests. :)